homeassistant/components/apache_kafka/__init__.py

```diff
@@ -141,7 +141,8 @@ async def shutdown(self, _: Event) -> None:
 
     async def write(self, event: Event[EventStateChangedData]) -> None:
         """Write a binary payload to Kafka."""
+        key = event.data["entity_id"].encode("utf-8")
         payload = self._encode_event(event)
 
-        if payload:
-            await self._producer.send_and_wait(self._topic, payload)
+        if key and payload:
```

Comment:
Since `key` is expected to always exist (given that you're doing a direct dict access above), do we need to check that it exists here?